### PR TITLE
WEEK 5

### PR DESCRIPTION
**Pull request by spartak0**

### CODE REVIEW

File: app/src/main/java/com/example/weather/ui/temp/TempFragment.java [hunk 1]
```diff
@@ -58,4 +93,11 @@ public void onClick(View view) {
             }
         });
     }
-}
\ No newline at end of file
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        viewModel.update();
```
Comment (YellowShark) [inline app/src/main/java/com/example/weather/ui/temp/TempFragment.java]: image

File: app/src/main/java/com/example/weather/ui/temp/TempViewModel.java [hunk 2]
```diff
@@ -55,11 +62,53 @@ public Completable deleteWeatherById(int id){
     @SuppressLint("CheckResult")
     public void fetchDailyWeather(int id) {
         getWeatherById(id).subscribe(weatherData -> {
-            getDailyWeatherByCoord(weatherData.getLan(), weatherData.getLon())
+            getDailyWeatherByCoord(weatherData.getLan(), weatherData.getLon(),0)
+                    .subscribe(hashMap -> {
+                        Map<String, String> tmp = liveData.getValue();
+                        tmp.put(Constant.MORN, hashMap.get(Constant.MORN_MAPPER));
```
Comment (YellowShark) [inline app/src/main/java/com/example/weather/ui/temp/TempViewModel.java]: I'd recommend to change this. It isn't scalable. What if we are going to display more weathers? Will we add more constants? I'd recommend to make a list.

File: app/src/main/res/layout/fragment_temp.xml [hunk 3]
```diff
@@ -59,43 +61,134 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:gravity="center_horizontal"
+                android:text="@string/night" />
+        </TableRow>
 
         <TableRow
             android:layout_width="match_parent"
-            android:layout_height="match_parent" />
+            android:layout_height="match_parent"
```
Comment (YellowShark) [inline app/src/main/res/layout/fragment_temp.xml]: So, why we can't use `RecyclerView` here?

**Review by YellowShark — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_temp.xml (hunk 3 above)
Comment (spartak0) [reply app/src/main/res/layout/fragment_temp.xml]: Why? We do not need scroll, these are static views for displaying the weather.

File: app/src/main/java/com/example/weather/ui/temp/TempFragment.java (hunk 1 above)
Comment (spartak0) [reply app/src/main/java/com/example/weather/ui/temp/TempFragment.java]: This is genius move for TempView!!! I'll explain it at the university

**Review by YellowShark — APPROVED** [review]

**Pull request merged**